tests/checked_set_test.py

```diff
@@ -71,6 +72,7 @@ def test_create():
 def test_evolver_returns_same_instance_when_no_updates():
     x = Naturals([1, 2])
     assert x.evolver().persistent() is x
+    assert x.evolver().add(1).update([1, 2]).persistent() is x
```

Comment:
Hmm. It seems that the evolver is considered dirty as soon as a new element is added so I'm going to leave this.

```
In [27]: Naturals([1,2]).evolver().add(1)
Out[27]: <pyrsistent._checked_types.Evolver at 0x7f07b6487dc0>

In [28]: Naturals([1,2]).evolver().add(1).is_dirty()
Out[28]: False

In [29]: Naturals([1,2]).evolver().add(3).is_dirty()
Out[29]: True

In [30]: Naturals([1,2]).evolver().add(3).remove(3).is_dirty()
Out[30]: True
```

And it seems to dispatch down to the C module

```
In [32]: e._pmap_evolver._buckets_evolver.is_dirty()
Out[32]: True

In [33]: e._pmap_evolver._buckets_evolver??
Type:        pvector_evolver
String form: <pvector_evolver object at 0x7f07b6483cf8>
Length:      8
Docstring:   <no docstring>


```